sdk/lib/core/date_time.dart

```diff
@@ -184,6 +184,41 @@ class DateTime implements Comparable {
           year, month, day, hour, minute, second, millisecond, true);
 
   /**
+   * Constructs a [DateTime] instance that contains the same data as the given instance,
+   * but provides named fields to overwrite fields with different values if necessary.
+   *
+   * For example,
+   * to create a new DateTime object representing April 29, 2014, 6:04am:
+   *
+   *     DateTime annularEclipse = new DateTime(2014, DateTime.APRIL, 29, 6, 4);
+   *
+   * Now you need a second DateTime on another Day, but the rest should stay the same
+   *
+   *     DateTime second = new DateTime.from(annularEclipse, month: DateTime.SEPTEMBER, day: 20);
+   */
+  DateTime.from(
+      DateTime src, {
+      int year : null,
```

Comment:
I think the normal style is to put no space before the ':' and one space after:
`int year: null,`